src/node_data.cpp

```diff
@@ -256,9 +256,21 @@ void node_data::reset_map() {
 }
 
 void node_data::insert_map_pair(node& key, node& value) {
-  m_map[&key] = &value;
+
+  bool found = false;
```

Comment:
Rereading this, I realized that this makes inserting into a map node a linear operation, which isn't good. The upshot, I think, is that we should always just append this to the list, and make a breaking change to force_insert so that it might duplicate a key when added twice that way.

Sorry for the churn.